### PR TITLE
Deployment fixes

### PR DESCRIPTION
**Pull request by radowit**

### CODE REVIEW

**Comment by hasparus** [comment]
Can we move `hasura` directory to the root?

File: docker-compose.local.yml [hunk 1]
```diff
@@ -8,9 +8,9 @@ services:
     ports:
       - 5432:5432
     environment:
-      - POSTGRES_USER=hello_django
```
Comment (lprasol) [inline docker-compose.local.yml]: Let's put it to the .env file, we have some of the environment variables here some there.

File: .env.prod.db-sample [hunk 2]
```diff
@@ -1,3 +1,3 @@
-POSTGRES_USER=hello_django
-POSTGRES_PASSWORD=hello_django
-POSTGRES_DB=hello_django_prod
+POSTGRES_USER=hello_zm
```
Comment (lprasol) [inline .env.prod.db-sample]: Proposition - use one .env file, we'll avoid mstakes and typos/

File: docker-compose.local.yml [hunk 3]
```diff
@@ -30,12 +30,12 @@ services:
     - postgres
     restart: always
     environment:
-      HASURA_GRAPHQL_DATABASE_URL: postgres://hello_django:hello_django@postgres:5432/hello_django_dev
+      HASURA_GRAPHQL_DATABASE_URL: postgres://hello_zm:hello_zm@postgres:5432/hello_zm_dev
```
Comment (lprasol) [inline docker-compose.local.yml]: to `.env` file?

File: docker-compose.prod.yml [hunk 4]
```diff
@@ -23,7 +23,7 @@ services:
     - db
     restart: always
     environment:
-      HASURA_GRAPHQL_DATABASE_URL: postgres://hello_django:hello_django@db:5432/hello_django_dev
+      HASURA_GRAPHQL_DATABASE_URL: postgres://hello_zm:hello_zm@db:5432/hello_zm_dev
```
Comment (lprasol) [inline docker-compose.prod.yml]: To `.env` file?

File: heroku/README.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+![](./assets/2020-03-21-15-59-31.png)
+
+## See the logs
+
+**https://dashboard.heroku.com/apps/zagrajmy-db-dev/logs**
+
+## Kill stuck builds
+
+```
+heroku plugins:install heroku-builds
```
Comment (lprasol) [inline heroku/README.md]: Is it up to datee?

File: heroku/deploy.sh [hunk 6]
```diff
@@ -0,0 +1,3 @@
+git subtree split --prefix packages/app/data/throwaway -b temporary
```
Comment (lprasol) [inline heroku/deploy.sh]: Do we use it?

**Comment by radowit** [comment]
> Can we move `hasura` directory to the root?

The hasura config here is strictly connected to heroku deployment

File: docker-compose.local.yml (hunk 1 above)
Comment (radowit) [reply docker-compose.local.yml]: ok

File: .env.prod.db-sample (hunk 2 above)
Comment (radowit) [reply .env.prod.db-sample]: ok

File: docker-compose.local.yml (hunk 3 above)
Comment (radowit) [reply docker-compose.local.yml]: ok

File: docker-compose.prod.yml (hunk 4 above)
Comment (radowit) [reply docker-compose.prod.yml]: ok

File: heroku/README.md (hunk 5 above)
Comment (radowit) [reply heroku/README.md]: dunno, ask @hasparus

File: heroku/deploy.sh (hunk 6 above)
Comment (radowit) [reply heroku/deploy.sh]: @hasparus

File: .env.dev [hunk 7]
```diff
@@ -1 +1,16 @@
+# Docker compose
+COMPOSE_FILE=docker-compose.local.yml
+
+# web
 DEBUG=1
+
+# graphql-engine
+HASURA_GRAPHQL_ADMIN_SECRET=dev_only_password
+HASURA_GRAPHQL_DATABASE_URL=postgres://zagrajmy:zagrajmy@postgres:5432/zagrajmy_dev
```
Comment (lprasol) [inline .env.dev]: If we moved postgres conf above, is it possible to to use $(POSTGRES_DB) etc here?

File: docker-compose.local.yml [hunk 8]
```diff
@@ -8,9 +8,9 @@ services:
     ports:
       - 5432:5432
     environment:
-      - POSTGRES_USER=hello_zm
-      - POSTGRES_PASSWORD=hello_zm
-      - POSTGRES_DB=hello_zm_dev
+      - POSTGRES_DB
```
Comment (lprasol) [inline docker-compose.local.yml]: I do not think we need it. It is in `.env`

File: docker-compose.local.yml [hunk 9]
```diff
@@ -30,10 +32,10 @@ services:
     - postgres
     restart: always
     environment:
-      HASURA_GRAPHQL_DATABASE_URL: postgres://hello_zm:hello_zm@postgres:5432/hello_zm_dev
-      HASURA_GRAPHQL_ENABLE_CONSOLE: "true" # set to "false" to disable console
-      HASURA_GRAPHQL_ENABLED_LOG_TYPES: startup, http-log, webhook-log, websocket-log, query-log
-      HASURA_GRAPHQL_ADMIN_SECRET: dev_only_password
+      - HASURA_GRAPHQL_ADMIN_SECRET
```
Comment (lprasol) [inline docker-compose.local.yml]: IMHO it may be a) not needed b) harmful if overrides `.env`

File: docker-compose.prod.yml [hunk 10]
```diff
@@ -9,12 +9,18 @@ services:
       - 8000
     env_file:
       - ./.env.prod
+    environment:
+      - DEBUG
```
Comment (lprasol) [inline docker-compose.prod.yml]: DEBUG on potentally production docker-compose?

File: docker-compose.prod.yml [hunk 11]
```diff
@@ -9,12 +9,18 @@ services:
...
   db:
     image: postgres:12.0-alpine
     volumes:
       - postgres_data:/var/lib/postgresql/data/
     env_file:
       - ./.env.prod.db
+    environment:
+      - POSTGRES_DB
```
Comment (lprasol) [inline docker-compose.prod.yml]: not needed if exist in `.env`

File: .env.dev (hunk 7 above)
Comment (radowit) [reply .env.dev]: ok

File: docker-compose.prod.yml [hunk 12]
```diff
@@ -23,10 +29,10 @@ services:
     - db
     restart: always
     environment:
-      HASURA_GRAPHQL_DATABASE_URL: postgres://hello_zm:hello_zm@db:5432/hello_zm_dev
-      HASURA_GRAPHQL_ENABLE_CONSOLE: "true" # set to "false" to disable console
-      HASURA_GRAPHQL_ENABLED_LOG_TYPES: startup, http-log, webhook-log, websocket-log, query-log
-      HASURA_GRAPHQL_ADMIN_SECRET: dev_only_password
+      - HASURA_GRAPHQL_ADMIN_SECRET
```
Comment (lprasol) [inline docker-compose.prod.yml]: not needed if exist in `.env`

File: docker-compose.local.yml (hunk 8 above)
Comment (radowit) [reply docker-compose.local.yml]: Docker isn't automatically passing all env variables to all containers

File: docker-compose.local.yml (hunk 9 above)
Comment (radowit) [reply docker-compose.local.yml]: Docker isn't automatically passing all env variables to all containers

File: docker-compose.prod.yml (hunk 10 above)
Comment (radowit) [reply docker-compose.prod.yml]: ok

File: docker-compose.prod.yml (hunk 11 above)
Comment (radowit) [reply docker-compose.prod.yml]: Docker isn't automatically passing all env variables to all containers

File: docker-compose.prod.yml (hunk 12 above)
Comment (radowit) [reply docker-compose.prod.yml]: Docker isn't automatically passing all env variables to all containers

File: heroku/README.md (hunk 5 above)
Comment (hasparus) [reply heroku/README.md]: this file mentions being moved to backend repo in the future :D it isn’t up to date

Comment (radowit) [reply heroku/README.md]: XD

File: .env.dev (hunk 7 above)
Comment (radowit) [reply .env.dev]: @lprasol 
Looks like inside `.env` we cannot reference other variables.

**Pull request merged**